core/src/main/java/io/fabric8/maven/core/service/kubernetes/jib/JibServiceUtil.java

```diff
@@ -162,18 +170,24 @@ public static void jibPush(ImageConfiguration imageConfiguration, MavenProject p
      * @param logger
      */
     private static void pushImage(TarImage baseImage, String targetImageName, Credential credential, Logger logger) {
+        long timeOut = Long.valueOf(EnvUtil.getEnvVarOrSystemProperty("fabric8.build.jib.timeOut", "60"));
+        SingleThreadedExecutor singleThreadedExecutor = new SingleThreadedExecutor();
+        consoleLogger = getConsoleLogger(logger, singleThreadedExecutor);
         try {
             RegistryImage targetImage = RegistryImage.named(targetImageName);
             if (credential!= null && !credential.getUsername().isEmpty() && !credential.getPassword().isEmpty()) {
                 targetImage.addCredential(credential.getUsername(), credential.getPassword());
             }
 
             Jib.from(baseImage).containerize(Containerizer.to(targetImage)
-                    .addEventHandler(LogEvent.class, log(logger))
-                    .addEventHandler(TimerEvent.class, new TimerEventHandler(logger::debug))
-                    .addEventHandler(ProgressEvent.class, new ProgressEventHandler(logUpdate(logger))));
+                    .addEventHandler(LogEvent.class, JibServiceUtil::log)
+                    .addEventHandler(TimerEvent.class, new TimerEventHandler(message -> consoleLogger.log(LogEvent.Level.DEBUG, message)))
+                    .addEventHandler(ProgressEvent.class, new ProgressEventHandler(update -> consoleLogger
+                            .setFooter(ProgressDisplayGenerator.generateProgressDisplay(update.getProgress(), update.getUnfinishedLeafTasks())))));
+            singleThreadedExecutor.shutDownAndAwaitTermination(Duration.ofSeconds(timeOut));
         } catch (RegistryException | CacheDirectoryCreationException | InvalidImageReferenceException | IOException | ExecutionException | InterruptedException e) {
```

Comment:
bdw, if you're shutting down your executor both in try and catch block; shouldn't you be doing it in finally?

Reply:
**Where is my mind by Pixies plays...**

Sorry that skipped my mind.